numpy/_core/src/multiarray/shape.h

```diff
@@ -1,6 +1,8 @@
 #ifndef NUMPY_CORE_SRC_MULTIARRAY_SHAPE_H_
 #define NUMPY_CORE_SRC_MULTIARRAY_SHAPE_H_
 
+#include "conversion_utils.h"
```

Comment:
usually we don't put includes in headers, can you move this to the implementation files where it's needed?

Reply:
I put this header here because `_reshape_with_copy_arg` declaration now contains `NPY_COPYMODE copy` that is defined in `conversion_utils.h`. Without it, it fails with:
```
error: unknown type name 'NPY_COPYMODE'
```
There are other header files with `includes`, such as `dtypemeta.h`.

Reply:
(left a comment I quickly realized was dumb and deleted, sorry for the noise)

Maybe it would be cleaner to define the new function in `conversion_utils.c`?